### PR TITLE
Prevent Activity from Recreating & Fixed Widget Crash

### PR DESCRIPTION
**Pull request by deplacer**
Activities are recreated when a user rotate the screen. 
Try to select conversation then rotate the screen. The selected conversation will gone. 
This will also fixed Force close Problem In Sub Settings page #561

### CODE REVIEW

**Comment by moezbhatti** [comment]
> android:configChanges="orientation|screenSize"

This isn't a great fix. Ideally the activities are set up so that they can properly be recreated. There are many ways that an activity can be recreated, so just because we're telling it not to recreate itself now, doesn't mean it won't in some other case, and then the app will still crash because the recreation isn't handled properly

File: QKSMS/src/main/java/com/moez/QKSMS/ui/widget/WidgetProvider.java [hunk 1]
```diff
@@ -88,8 +90,14 @@ public void onAppWidgetOptionsChanged(Context context, AppWidgetManager appWidge
...
-        Bundle options = appWidgetManager.getAppWidgetOptions(appWidgetId);
-        int size = options.getInt(AppWidgetManager.OPTION_APPWIDGET_MIN_WIDTH);
+        int size;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            Bundle options = appWidgetManager.getAppWidgetOptions(appWidgetId);
+            size = options.getInt(AppWidgetManager.OPTION_APPWIDGET_MIN_WIDTH);
+        } else {
+            AppWidgetProviderInfo appWidgetInfo = appWidgetManager.getAppWidgetInfo(appWidgetId);
+            size = appWidgetInfo.minWidth;
+        }
```
Comment (moezbhatti) [inline QKSMS/src/main/java/com/moez/QKSMS/ui/widget/WidgetProvider.java]: Looks good

**Comment by deplacer** [comment]
Ah.. you're right.. it might fixed some issues but in reality more issue might exist. 
The main problem was on SettingsActivity. Here is the exception when you go to Settings > click any Sub Settings (for instance Appearance) > Rotate the screen then press Back.

```
    @Override
    public void onBackPressed() {
        if (getFragmentManager().getBackStackEntryCount() > 0) {
            getFragmentManager().popBackStack();
        } else {
            super.onBackPressed();
        }
    }
```
onBackPressed() throws an exception.

```
                                                                Process: com.moez.QKSMS, PID: 28447
                                                                java.lang.IllegalStateException: Fragment already added: SettingsFragment{ad7d4810 #0 id=0x7f100095 SettingsFragment}
                                                                    at android.app.FragmentManagerImpl.addFragment(FragmentManager.java:1133)
                                                                    at android.app.BackStackRecord.popFromBackStack(BackStackRecord.java:725)
                                                                    at android.app.FragmentManagerImpl.popBackStackState(FragmentManager.java:1496)
                                                                    at android.app.FragmentManagerImpl$2.run(FragmentManager.java:486)
                                                                    at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1447)
                                                                    at android.app.FragmentManagerImpl$1.run(FragmentManager.java:443)
                                                                    at android.os.Handler.handleCallback(Handler.java:733)
                                                                    at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                    at android.os.Looper.loop(Looper.java:136)
                                                                    at android.app.ActivityThread.main(ActivityThread.java:5017)
                                                                    at java.lang.reflect.Method.invokeNative(Native Method)
                                                                    at java.lang.reflect.Method.invoke(Method.java:515)
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
                                                                    at dalvik.system.NativeStart.main(Native Method)
```

**Comment by moezbhatti** [comment]
Can confirm, this does cause a crash. I can look into that this week

File: QKSMS/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -484,6 +484,11 @@
                 <action android:name="com.mariussoft.endlessjabber.action.extend" />
             </intent-filter>
         </receiver>
+        <receiver android:name=".receiver.AirplaneModeReceiver">
+            <intent-filter>
+                <action android:name="android.intent.action.AIRPLANE_MODE" />
+            </intent-filter>
+        </receiver>
```
Comment (moezbhatti) [inline QKSMS/src/main/AndroidManifest.xml]: The changes to this file can be reverted

Comment (deplacer) [inline QKSMS/src/main/AndroidManifest.xml]: Hmm.. cannot figure out how to revert this changes..

Comment (moezbhatti) [inline QKSMS/src/main/AndroidManifest.xml]: You can just manually erase what was added, and re-insert what was removed
